### ISSUE TITLE
[HOLD for payment 2023-06-15] [$1000] Deleted message gets highlighted when we send a message, delete the message, visit any other chat and revisit the chat in offline mode

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#18165]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app
2. Open any report
3. Switch to offline mode
4. Delete any message (currently it is not highlighted)
5. Switch to other report and revisit the deleted message report to see that message is highlighted now
6. Switch off offline mode
7. Again visit the chat to see that deleted message is now gone
8. Switch back the offline mode
9. Revisit the chat and deleted message is back and it is still highlighted

## Expected Result: 
App should not highlight deleted message on revisiting chat in offline mode nor have the deleted message reappear after reconnecting online then offline
## Actual Result: 
App highlights deleted message on revisiting chat in offline mode and highlight remains intact. App shows deleted message in offline mode even after going back online then offline again.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.8
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43996225/235247949-156727ea-9f12-4048-a81e-55280d5f022e.mp4

https://user-images.githubusercontent.com/43996225/235247965-22b4082c-8cec-4b49-b0ef-286b729d9340.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1682681346161399

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013afb44ff7c31e750</li>
        <li>Upwork Job ID: 1654091952987746304</li>
        <li>Last Price Increase: 2023-06-01</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @dylanexpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by dylanexpensify** [comment]
reviewing today!

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The deleted message while offline has a hover effect after switching back from another chat.

### What is the root cause of that problem?
The hover effect will show if:
1. is hovered
2. is a whisper
3. a context menu active
4. is a draft message

and the condition that it meets is the third one. Why?

When we delete a chat, it will show a delete modal by calling `showDeleteModal` of `PopoverReportActionContextMenu` passing the selected report action id and the report id.
https://github.com/Expensify/App/blob/ce3b938b495b28fed058a6c2f1d50bb2aac7961c/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L287-L296

When we confirm the deletion, we 
1. delete the message
2. hide the modal
https://github.com/Expensify/App/blob/ce3b938b495b28fed058a6c2f1d50bb2aac7961c/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L261-L265

One thing is missing is to reset the report action id and report id. So, when we reopen the chat, it will remount the component and check, is this chat message has an active context menu?
https://github.com/Expensify/App/blob/ce3b938b495b28fed058a6c2f1d50bb2aac7961c/src/pages/home/report/ReportActionItem.js#L99-L101
https://github.com/Expensify/App/blob/ce3b938b495b28fed058a6c2f1d50bb2aac7961c/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L110-L112

Because we didn't clear the report action id, it will say, yes, a context menu is active for this chat message.

This issue however doesn't happen when we simply cancel the deletion because when we cancel it, we will call `hideDeleteModal` that will reset the report action id and along with other state.
https://github.com/Expensify/App/blob/ce3b938b495b28fed058a6c2f1d50bb2aac7961c/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L267-L277

### What changes do you think we should make in order to solve the problem?

Reset the other state (only report action id and report id is necessary) when we confirm the deletion (in `confirmDeleteAndHideModal`).

**Comment by dylanexpensify** [comment]
Didn't get to this today due to focusing on ECUK and CPM tasks! Will review tomorrow!

**Comment by dylanexpensify** [comment]
Posted about this to get clarification on expected behavior. Will update this once we come to a conclusion.

**Comment by dylanexpensify** [comment]
After discussing we confirmed that this is an expected behavior! @dhanashree-sawant [here's](https://github.com/Expensify/App/blob/main/contributingGuides/OFFLINE_UX.md#b---optimistic-with-feedback-pattern) where you can find more info!

**Comment by dhanashree-sawant** [comment]
Hi @dylanexpensify, In the document, we have mentioned that it should be greyed out but in the issue, the deleted message gets highlighted not greyed out. It gets greyed out when we delete but when we revisit the chat, it gets highlighted.

**Comment by dylanexpensify** [comment]
I believe that makes sense it's still highlighted while in offline mode, since technically, the delete action hasn't registered 100% yet. I think this would be a bug if we got back online, and the deleted comment did finish deletion, but then still highlighted after switching back. Mind confirming for me it doesn't do that?

**Comment by dhanashree-sawant** [comment]
Yes so on going back offline, the deleted message reappears and it is still highlighted, steps I did:
1. switched to offline mode
2. delete any message (currently it is not highlighted)
3. switch to other report and revisit the deleted message report to see that message is highlighted now
4. Switch off offline mode
5. Again visit the chat to see that deleted message is now gone
6. Switch back the offline mode
7. Revisit the chat and deleted message is back and it is still highlighted

**Comment by dylanexpensify** [comment]
And after step 5, is the message now gone? Did you show those steps in the video above?

**Comment by dhanashree-sawant** [comment]
No message still exist, sorry I haven't shown that part in the video , I will add new video below with step 6 and 7 shown in it.

**Comment by dylanexpensify** [comment]
Ah that'd be great if you can show the steps you just commented in a video! Thanks!

**Comment by dhanashree-sawant** [comment]
Here is the video

https://user-images.githubusercontent.com/87513216/236191245-aa446e79-970e-4912-8ef8-3e79b2de674f.mp4

**Comment by dylanexpensify** [comment]
@dhanashree-sawant that new video is super helpful, ty!
@bernhardoj appreciate the thoughts, and like where your head is at with it!

**Comment by dylanexpensify** [comment]
Was able to successfully repro. Thanks for the back and forth here team!

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~013afb44ff7c31e750

**Comment by MelvinBot** [comment]
Current assignee @dylanexpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @Santhosh-Sellavel (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @flodnv (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by dylanexpensify** [comment]
Updated the OP

**Comment by bernhardoj** [comment]
> deleted message reappear after reconnecting online then offline

@dylanexpensify Just re-read the updated expected result. I think this is a different issue. We previously have that issue here https://github.com/Expensify/App/issues/15048 which we decided to not handle it https://expensify.slack.com/archives/C01GTK53T8Q/p1680025815238389?thread_ts=1679595548.409089&cid=C01GTK53T8Q

**Comment by dylanexpensify** [comment]
Repened!

**Comment by maddylewis** [comment]
@dylanexpensify - im a bit lost on what the next step(s) on this issue need to be. going forward, please leave some context on what the new assignee should be doing on the GH (more info [here](https://stackoverflowteams.com/c/expensify/questions/8836#:~:text=what%20the%20next%20steps%20on%20the%20issue%20might%20be%2C%20and%20any%20other%20context%20the%20new%20assignee%20may%20need)).

**Comment by maddylewis** [comment]
@Santhosh-Sellavel or @flodnv - apologies, i was assigned this issue to cover for dylan while he's OOO. can someone give me an update on the status of this bug? are we still waiting on a proposal for this one?

**Comment by flodnv** [comment]
Actually I think we already have a proposal here: https://github.com/Expensify/App/issues/18165#issuecomment-1531179702

@Santhosh-Sellavel can you review it please?

**Comment by Santhosh-Sellavel** [comment]
Sorry, I missed that one will try to review that today.

**Comment by Santhosh-Sellavel** [comment]
Just For quick reference.

> ## Proposal
> ### Please re-state the problem that we are trying to solve in this issue.
> The deleted message while offline has a hover effect after switching back from another chat.
> 
> ### What is the root cause of that problem?
> The hover effect will show if:
> 
> 1. is hovered
> 2. is a whisper
> 3. a context menu active
> 4. is a draft message
>    https://github.com/Expensify/App/blob/ce3b938b495b28fed058a6c2f1d50bb2aac7961c/src/pages/home/report/ReportActionItem.js#L282-L286
> 
> and the condition that it meets is the third one. Why?
> 
> When we delete a chat, it will show a delete modal by calling `showDeleteModal` of `PopoverReportActionContextMenu` passing the selected report action id and the report id.
> 
> https://github.com/Expensify/App/blob/ce3b938b495b28fed058a6c2f1d50bb2aac7961c/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L287-L296
> 
> When we confirm the deletion, we
> 
> 1. delete the message
> 2. hide the modal
>    https://github.com/Expensify/App/blob/ce3b938b495b28fed058a6c2f1d50bb2aac7961c/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L261-L265
> 
> One thing is missing is to reset the report action id and report id. So, when we reopen the chat, it will remount the component and check, is this chat message has an active context menu?
> 
> https://github.com/Expensify/App/blob/ce3b938b495b28fed058a6c2f1d50bb2aac7961c/src/pages/home/report/ReportActionItem.js#L99-L101
> 
> 
> https://github.com/Expensify/App/blob/ce3b938b495b28fed058a6c2f1d50bb2aac7961c/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L110-L112
> 
> Because we didn't clear the report action id, it will say, yes, a context menu is active for this chat message.
> 
> This issue however doesn't happen when we simply cancel the deletion because when we cancel it, we will call `hideDeleteModal` that will reset the report action id and along with other state.
> 
> https://github.com/Expensify/App/blob/ce3b938b495b28fed058a6c2f1d50bb2aac7961c/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L267-L277
> 
> ### What changes do you think we should make in order to solve the problem?
> Reset the other state (only report action id and report id is necessary) when we confirm the deletion (in `confirmDeleteAndHideModal`).
> 
> ### What alternative solutions did you explore? (Optional)
> We can reset the report action id and report id in `onModalHide` callback of `ConfirmModal`
> 
> https://github.com/Expensify/App/blob/b6eb637298f77156c9dc9c482e30d77b105d2a26/src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js#L332

**Comment by Santhosh-Sellavel** [comment]
Looks good, but will revisit it tomorrow for the final call!

**Comment by Santhosh-Sellavel** [comment]
@bernhardoj This happens even when we cancel the delete popup and switch and come back to the chat again. So we should handle this too.

 Please include a snippet, or feel free to include diff links here if it requires.

https://github.com/Expensify/App/assets/85645967/cc3649ee-cd31-4d4a-b862-d0cc16fa0e7f

**Comment by bernhardoj** [comment]
@Santhosh-Sellavel we have a PR that remove the reportAction reset when we cancel the modal https://github.com/Expensify/App/pull/17348/files#r1165760995, but the solution to this issue is to reset it. The reason behind this can be read from the link.

I was planning to reset the reportAction when the modal completely hidden with `onModalHide` callback, but currently I'm unable to delete a money request to make sure it works as expected (the modal text does not change while hiding).

<img width="882" alt="image" src="https://github.com/Expensify/App/assets/50919443/9b6cc5b5-6801-480c-80b7-05e231b87e1a">

Notice no delete button in the context menu. I think this is a bug because the [PR](https://github.com/Expensify/App/pull/17348) expect a delete button to be present for unsettled money request.

**Comment by Santhosh-Sellavel** [comment]
@bernhardoj 

Bit confused. What changes are you exactly proposing here?

**Comment by bernhardoj** [comment]
@Santhosh-Sellavel Sorry if it confuses you. My current proposal is the changes I propose. My previous comment:

1. The first part is explaining why the issue also happens when cancelling the modal
2. The rest explaining that we need to fix a bug (or maybe it's intentional?) where a delete button is not present for money request action first. Why? To test and make sure our changes here does not create another issue that was fixed here https://github.com/Expensify/App/pull/17348/files#r1165760995

**Comment by Santhosh-Sellavel** [comment]
@bernhardoj 

Can you include or share code snippets for the changes your are proposing?

**Comment by bernhardoj** [comment]
Added in my proposal

**Comment by flodnv** [comment]
@Santhosh-Sellavel what do you think about the proposal?

**Comment by Santhosh-Sellavel** [comment]
Sorry for the delay, @bernhardoj proposal addressed the highlight issue mentioned

> App should not highlight deleted message on revisiting chat in offline mode nor have the deleted message reappear after reconnecting online and then offline



@flodnv I think we can hire @bernhardoj here, only highlighting issue that needs to be fixed here. As we used the force offline option, which cause unnecessary issues as API would have succeeded the first time itself, we just throw an error on the UI for testing purposes. In a real offline use case, the deleted messages are re-appearing again, but if there is enough time online, it goes away when the API succeeds.

**Comment by flodnv** [comment]
Thanks 👍 Go for it @bernhardoj

**Comment by dylanexpensify** [comment]
Upwork is down, once back up I'll send payment!!

**Comment by dylanexpensify** [comment]
We're back! @Santhosh-Sellavel @bernhardoj @dhanashree-sawant can you all please apply [here](https://www.upwork.com/jobs/~01358aa1453e1ed54e)! 🙇‍♂️

**Comment by dhanashree-sawant** [comment]
Hi @dylanexpensify, thanks I have applied to the job.

**Comment by dylanexpensify** [comment]
Offers sent, payments coming!

**Comment by dylanexpensify** [comment]
@Santhosh-Sellavel can we work on checklist today before we close out? 🙏

**Comment by dylanexpensify** [comment]
Commented above, Mel!

**Comment by dylanexpensify** [comment]
@Santhosh-Sellavel can you please apply? 🙇‍♂️

**Comment by dylanexpensify** [comment]
@Santhosh-Sellavel bumping

**Comment by Santhosh-Sellavel** [comment]
Will look into the checklist this week!

**Comment by dylanexpensify** [comment]
@Santhosh-Sellavel can we get an update here

**Comment by dylanexpensify** [comment]
@Santhosh-Sellavel let's get this done today 🙌

**Comment by Santhosh-Sellavel** [comment]
@bernhardoj Could use your help here is the https://github.com/Expensify/App/pull/17348/files#r1165760995 which caused this regression?

**Comment by dylanexpensify** [comment]
@Santhosh-Sellavel let's get an update here today please! 🙇‍♂️

**Comment by dylanexpensify** [comment]
Reviewing today!

**Comment by dylanexpensify** [comment]
Closing!

**Comment by flodnv** [comment]
Ah I forgot that was a thing. Maybe edge case makes sense yeah..

**Comment by Santhosh-Sellavel** [comment]
I've had not received payment for this @dylanexpensify will request via NewDot

**Comment by Santhosh-Sellavel** [comment]
Requested Payment on ND

**Comment by dylanexpensify** [comment]
reopened while we wait for @Santhosh-Sellavel to be paid! Cc @anmurali

**Comment by anmurali** [comment]
Approved $1500 (with bonus) for Santhosh
